plutus-tx-plugin/src/PlutusTx/Compiler/Expr.hs

```diff
@@ -403,12 +403,20 @@ compileExpr e = withContextM 2 (sdToTxt $ "Compiling expr:" GHC.<+> GHC.ppr e) $
     CompileContext {ccScopes=stack,ccBuiltinNameInfo=nameInfo} <- ask
 
     -- TODO: Maybe share this to avoid repeated lookups. Probably cheap, though.
-    (stringTyName, sbsName) <- case (Map.lookup ''Builtins.String nameInfo, Map.lookup 'String.stringToBuiltinString nameInfo) of
-        (Just t1, Just t2) -> pure $ (GHC.getName t1, GHC.getName t2)
-        _                  -> throwPlain $ CompilationError "No info for String builtin"
+    (stringTyName, sbsName, bsTyName) <- case
+        ( Map.lookup ''Builtins.String nameInfo
+        , Map.lookup 'String.stringToBuiltinString nameInfo
+        , Map.lookup ''Builtins.ByteString nameInfo
+        ) of
+            (Just t1, Just t2, Just t3) -> pure $ (GHC.getName t1, GHC.getName t2, GHC.getName t3)
+            _                           -> throwPlain $ CompilationError "No info for String builtin"
 
     let top = NE.head stack
     case e of
+        -- 'fromString' invocation at the builtin ByteString type
+        (strip -> GHC.Var (GHC.idDetails -> GHC.ClassOpId cls)) `GHC.App` GHC.Type (GHC.tyConAppTyCon_maybe -> Just tc) `GHC.App` _ `GHC.App` (strip -> stringExprContent -> Just bs)
+            | GHC.getName cls == GHC.isStringClassName, GHC.getName tc == bsTyName -> do
```

Comment:
To save duplicating the huge match, maybe branch on what `GHC.getName tc` inside?

The downside is that we won't fall through if we see a `fromString` call that's _not_ at one of the types we can handle. I don't know if that's a big deal or not. The advantage would be that we will just give up quickly if we see some other `fromString`. But the downside is that we'll give up... even if it would work! So I'm not sure what to do actually, but I'll leave this comment in case anyone has thoughts.